### PR TITLE
[consensus] Start using libra_channel for network <=> consensus event processor

### PR DESCRIPTION
**Pull request by ankushagarwal**
## Summary

Refactor `network.rs` to start using `libra_channel` between network <=> consensus event processor

## Test Plan

Existing unit tests

### CODE REVIEW

File: consensus/src/chained_bft/network.rs [hunk 1]
```diff
@@ -126,11 +129,26 @@ impl ConsensusNetworkImpl {
 
     /// Establishes the initial connections with the peers and returns the receivers.
     pub fn start<T: Payload>(&mut self, executor: &TaskExecutor) -> NetworkReceivers<T> {
-        let (proposal_tx, proposal_rx) = channel::new(1_024, &counters::PENDING_PROPOSAL);
-        let (vote_tx, vote_rx) = channel::new(1_024, &counters::PENDING_VOTES);
-        let (block_request_tx, block_request_rx) =
-            channel::new(1_024, &counters::PENDING_BLOCK_REQUESTS);
-        let (sync_info_tx, sync_info_rx) = channel::new(1_024, &counters::PENDING_SYNC_INFO_MSGS);
+        let (proposal_tx, proposal_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::FIFO,
```
Comment (dmitri-perelman) [inline consensus/src/chained_bft/network.rs]: Didn't we want to use LIFO in these cases?

**Comment by dmitri-perelman** [comment]
Makes sense! We cannot land it until we make sure there are no regressions based on some experiments on your local cluster.
@bors-libra delegate+

File: consensus/src/chained_bft/network.rs [hunk 2]
```diff
@@ -283,11 +286,26 @@ impl<T: Payload> NetworkTask<T> {
...
+            Some(&counters::PROPOSAL_DROPPED_MSGS),
+        ));
+        let (vote_tx, vote_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::LIFO,
+            1,
+            Some(&counters::VOTES_DROPPED_MSGS),
+        ));
+        let (block_request_tx, block_request_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::LIFO,
+            1,
```
Comment (phlip9) [inline consensus/src/chained_bft/network.rs]: I wonder if a max queue size of 1 is a little big aggressive. are we sure there are no cases where someone might send us, e.g., multiple concurrent block retrieval requests? there are also few freshness constraints on a block retrieval request, apart from the timeout...

**Pull request closed without merging**